feat: support comments

rise-erpelding opened the pull request:
## Description

The original statement about comments/pingbacks/trackbacks in the issue stated that "The 'Discussion' section with comments/pingbacks shows up for pages and posts, but checking the boxes doesn't change any behavior". After some discussion, we determined that we could try to temporarily remove all menu options for comments, trackbacks, and pingbacks, and possibly address at least comments later. Removing comments altogether (particularly from the default post) proved difficult and didn't seem like a worthwhile pursuit since it was intended to be a temporary solution only, so this work:
- removes pingbacks and trackbacks to the extent possible
- supports the ability to have comments on posts if desired

<!-- If using GitHub issues, set the issue number to close it on merge -->
Closes #75

<!-- If using external project management, link to the issue/specification -->
<!-- [Issue](https://example.com/ISSUE_NUMBER) -->

## To Validate

<!-- Add steps a reviewer should follow to validate your changes -->

1. Make sure all PR Checks have passed
2. Pull down this branch
3. Run `npm start`
4. View the "Hello world!" default post and confirm that the default comment (by "A WordPressCommenter" appears)
5. Check nested comments and ability to add comments:
    - You should be able to add a new comment
    - Adding a new comment should add another list item to the comments list with the comment
    - You should be able to reply to the existing comment, or reply to a new comment
    - You should be able to reply to a comment reply
    - Replying to a comment should nest that comment under the comment or comment reply to which you are applying
    - You should be able to add a comment when logged out, and see that this comment shows up for approval in the [Comments section](http://localhost:8000/wp-admin/edit-comments.php)
6. Other links in Comments section appear to function as expected.
7. Checking and unchecking "Show Avatars" in [Discussion Settings](http://localhost:8000/wp-admin/options-discussion.php) appears to have an effect on whether or not avatars appear in comments section (don't forget to Save Changes!)
8. Check that password-protected posts do not show comments unless you are viewing the post:
    - Add comments to an existing post, 
9. Toggling "Allow comments" on a post, or "Allow people to submit comments on new posts" from the [Discussion Settings](http://localhost:8000/wp-admin/options-discussion.php) (don't forget to Save Changes!) seems to have an effect on whether or not you should be able to add a comment.
10. Toggling "Allow comments" off on an existing post with comments shows existing comments along with a message that says that comments have been closed. Comment form does not appear.
11. Container does not abruptly stop in the process of doing any of this (this was happening to me, but I believe it was the result of making changes to `comments.php`)

rise-erpelding commented on src/php/views/page.twig:
```diff
@@ -1,9 +1,10 @@
 	<div class="obj-width-limiter">
 		{{ post.content }}
+		{# Loads comments.php by default #}
+		{{ function('comments_template') }}
```
This seemed like the easiest way to make the default comment functionality in WordPress *just work*, and in fact, comments should display when this is added even without `comments.php` (or `comments.twig`), but it will give a warning that it doesn't like this. I'm a little wary of using this because it feels a little different than how we're including the rest of our templates, so I'm very happy to have any kind of feedback on this.

dustin-jw replied:
I think this is fine, we include other WordPress-y templates in a similar way, like the search and password forms, so this isn't a huge departure.

rise-erpelding commented on src/php/inc/theme-setup.php:
```diff
@@ -95,3 +95,50 @@ function custom_hide_editor() {
+ * Remove Trackbacks Support for the "post" Post Type.
+ */
+function remove_comments_support() {
+	remove_post_type_support( 'post', 'trackbacks' );
```
You would think that if you can remove this checkbox from the post, that you could remove checkboxes from the Discussion Settings as well, but I didn't find a way.

dustin-jw replied:
I found a snippet that seems to do the trick, but it does like 10 different things, so it might be a little heavy-handed. We could include it and maybe make it depend on an environment variable, like only run it if `COMMENTS_ENABLED=false`. Another option would be to document how to turn off comments for all future posts through the admin, and include that in basic setup instructions, kind of like we describe how to change the permalinks. Anyway, here's that snippet:

```php
add_action('admin_init', function () {
    // Redirect any user trying to access comments page
    global $pagenow;

    if ($pagenow === 'edit-comments.php') {
        wp_safe_redirect(admin_url());
        exit;
    }

    // Remove comments metabox from dashboard
    remove_meta_box('dashboard_recent_comments', 'dashboard', 'normal');

    // Disable support for comments and trackbacks in post types
    foreach (get_post_types() as $post_type) {
        if (post_type_supports($post_type, 'comments')) {
            remove_post_type_support($post_type, 'comments');
            remove_post_type_support($post_type, 'trackbacks');
        }
    }
});

// Close comments on the front-end
add_filter('comments_open', '__return_false', 20, 2);
add_filter('pings_open', '__return_false', 20, 2);

// Hide existing comments
add_filter('comments_array', '__return_empty_array', 10, 2);

// Remove comments page in menu
add_action('admin_menu', function () {
    remove_menu_page('edit-comments.php');
});

// Remove comments links from admin bar
add_action('init', function () {
    if (is_admin_bar_showing()) {
        remove_action('admin_bar_menu', 'wp_admin_bar_comments_menu', 60);
    }
});
```

rise-erpelding commented on src/php/comments.php:
```diff
@@ -13,12 +13,3 @@
 
 Timber\Timber::render( 'partials/comments.twig', $context );
-
-/*
```
This was already here on an existing `comments.php` template I found, but it doesn't seem like we need it.

rise-erpelding commented on src/php/comments.php:
```diff
@@ -13,12 +13,3 @@
- * the visitor has not yet entered the password,
- * return early without loading the comments.
- */
-if ( post_password_required() ) {
```
This was already here on an existing comments.php template I found, but it doesn't seem like we need it.

dustin-jw commented on src/php/inc/theme-setup.php:
```diff
@@ -95,3 +95,50 @@ function custom_hide_editor() {
+/**
+ * Remove Trackbacks Support for the "post" Post Type.
+ */
+function remove_comments_support() {
```
Since we're not actually removing comments support, we should rename this to `remove_trackbacks_support`.

rise-erpelding commented:
@dustin-jw I updated documentation with more how-to regarding comments. Looking over #75 I think your latest documentation update (to address #91) adequately addresses the item "It's unclear how to use custom taxonomies and whether they're working correctly", so this would be the last PR related to #75, therefore I updated the PR to close the issue when merged.

dustin-jw reviewed (approved):
This looks great, thanks @rise-erpelding!
